homeassistant/components/remote/kira.py

```diff
@@ -0,0 +1,85 @@
+"""
+Support for Keene Electronics IR-IP devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/remote.kira/
+"""
+import logging
+import functools as ft
+
+import homeassistant.components.remote as remote
+from homeassistant.helpers.entity import Entity
+from homeassistant.loader import get_component
+
+from homeassistant.const import (
+    STATE_UNKNOWN,
+    CONF_NAME)
+
+REQUIREMENTS = []
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Kira platform."""
+    if discovery_info:
+        name = discovery_info.get(CONF_NAME)
+
+        add_devices([KiraRemote(name)])
+    return True
+
+
+class KiraRemote(Entity):
+    """Remote representation used to send commands to a Kira device."""
+
+    def __init__(self, name):
+        """Initialize KiraRemote class."""
+        _LOGGER.debug("KiraRemote device init started for: %s", name)
+        self._name = name
+        self._state = STATE_UNKNOWN
+
+        kira = get_component('kira')
+        self._kira = kira.MODULES[name]
+
+    @property
+    def name(self):
+        """Return the Kira device's name."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Add platform specific attributes."""
+        return {}
+
+    @property
+    def is_on(self):
+        """Return True. Power state doesn't apply to this device"""
+        return True
+
+    def update(self):
+        """No-op."""
+
+    def turn_on(self, **kwargs):
+        """No-op."""
+
+    def turn_off(self):
```

Comment:
No need to overwrite this if you already overwrite `async_turn_off`. (which is faster so better.) Same for `turn_on`